### PR TITLE
Docker: adding use of bootstrap file

### LINKED ISSUE #116: Dockerfile: use bootstrap call in Dockerfile
The use of the boostrap script in the Dockerfile was removed because we need to pass ``--unsafe`` or ``--user=root``. See #91 

Instead probably the boostrap script should be fixed to allow for this. See discussion https://github.com/inveniosoftware/cookiecutter-invenio-instance/pull/91/files#r241034765

### PR DESCRIPTION
**Pull request by roksys**
* Closes #116

Co-authored-by: Georgios Lignos georgios.lignos@cern.ch

### CODE REVIEW

File: {{cookiecutter.project_shortname}}/Dockerfile.base [hunk 1]
```diff
@@ -2,4 +2,4 @@
 FROM inveniosoftware/centos7-python:3.6
 
 COPY Pipfile Pipfile.lock ./
-RUN pipenv install --deploy
+RUN pipenv install --deploy --system
```
Comment (lnielsen) [inline {{cookiecutter.project_shortname}}/Dockerfile.base]: What's the effect of ``--system``?

File: {{cookiecutter.project_shortname}}/scripts/bootstrap [hunk 2]
```diff
@@ -6,38 +6,60 @@ set -e
...
+  pipenv sync --dev
+  # Install application code and entrypoints from 'setup.py'
+  pipenv run pip install -e $script_path/..
+  # Build assets
+  pipenv run invenio collect -v
+  pipenv run invenio webpack buildall
+elif [ "$flags" == "--deploy" ]; then
+  check_for_pipfilelock
+  pipenv install --system --deploy
+  # Install application code and entrypoints from 'setup.py'
```
Comment (lnielsen) [inline {{cookiecutter.project_shortname}}/scripts/bootstrap]: I don't think we should be using ``-e`` for production? (i know it's coming from the previous)

File: {{cookiecutter.project_shortname}}/docker-compose.full.yml [hunk 3]
```diff
@@ -30,7 +30,7 @@ services:
     extends:
       file: docker-services.yml
       service: app
-    command: pipenv run uwsgi /opt/invenio/var/instance/uwsgi_ui.ini
+    command: ["uwsgi /opt/invenio/var/instance/uwsgi_ui.ini"]
```
Comment (lnielsen) [inline {{cookiecutter.project_shortname}}/docker-compose.full.yml]: Doesn't this just circumvent the pipenv?

File: {{cookiecutter.project_shortname}}/Dockerfile.base (hunk 1 above)
Comment (roksys) [reply {{cookiecutter.project_shortname}}/Dockerfile.base]: It installs dependencies outside virtualenv.

File: {{cookiecutter.project_shortname}}/docker-compose.full.yml (hunk 3 above)
Comment (roksys) [reply {{cookiecutter.project_shortname}}/docker-compose.full.yml]: Yes, @slint suggested to install all dependencies directly to system, not to virtualenv. That's why `pipenv install` has `--sytem` flag.

**Comment by slint** [comment]
@roksys I guess this can be closed? (if, yes close it)

**Pull request closed without merging**